scripts/makeGitTag.ts

```diff
@@ -0,0 +1,119 @@
+import fs from 'node:fs'
+import path from 'node:path'
+
+import { spawn } from 'cross-spawn'
+import yargs from 'yargs/yargs'
+
+class Shell {
+  stdout: string
+  stderr: string
+  returncode: number
+  cmd: string
+  constructor(cmd: string, args: string[], verbose = true) {
+    this.cmd = `${cmd} ${args.join(' ')}`
+    if (verbose) {
+      process.stderr.write(`${this.cmd}\n`)
+    }
+    const child = spawn.sync(cmd, args)
+    this.stdout =
+      child.stdout === null ? '' : String.fromCodePoint(...child.stdout)
+    this.stderr =
+      child.stderr === null ? '' : String.fromCodePoint(...child.stderr)
+    this.returncode = child.status
+
+    if (this.returncode != 0) {
+      throw new Error(
+        `\n${this.cmd}\nSTDOUT:\n${this.stdout}\nSTDERR:\n${this.stderr}\nEXIT CODE: ${this.returncode}`,
+      )
+    }
+  }
+}
+
+function getPackages(): string[] {
+  const p = new Shell('yarn', ['workspaces', 'list', '--json', '--no-private'])
+  const lines: string[] = p.stdout.trim().split('\n')
+  const packages: string[] = []
+  for (const line of lines) {
+    const j = JSON.parse(line)
+    const pack: string = path.join(j.location, 'package.json')
+    if (!fs.existsSync(pack)) {
+      throw new Error(`Error: Expected file '${pack}' does not exist\n`)
+    }
+    packages.push(pack)
+  }
+  return packages
+}
+
+function checkTag(tag: string, testRegex = /v\d+.\d+.\d+.*/) {
+  if (!testRegex.test(tag)) {
+    throw new Error(`Invalid tag: '${tag}' does not match regex ${testRegex}\n`)
+  }
+}
+
+function updatePackageVersion(newVersion: string, packagesToUpdate: string[]) {
```

Comment:
This should update all packages. Even if the private packages aren't getting published, their versions should still be updated. Also, it should use `yarn workspaces foreach` instead of calling it in each directory separately.